test/gurka/test_closure_annotations.cc

```diff
@@ -0,0 +1,327 @@
+#include "gurka.h"
+#include "test.h"
+
+#include <boost/format.hpp>
+#include <gtest/gtest.h>
+
+using namespace valhalla;
+using LiveTrafficCustomize = test::LiveTrafficCustomize;
+
+namespace {
+
+inline void SetLiveSpeedFrom(baldr::TrafficSpeed* live_speed, uint64_t speed, uint8_t subsegment) {
+  live_speed->breakpoint1 = subsegment;
+  live_speed->breakpoint2 = 255;
+  live_speed->speed2 = speed >> 1;
+  live_speed->speed3 = UNKNOWN_TRAFFIC_SPEED_RAW;
+}
+
+inline void SetLiveSpeedUpto(baldr::TrafficSpeed* live_speed, uint64_t speed, uint8_t subsegment) {
+  live_speed->breakpoint1 = subsegment;
+  live_speed->speed1 = speed >> 1;
+}
+
+inline void SetLiveSpeed(baldr::TrafficSpeed* live_speed, uint64_t speed) {
+  live_speed->breakpoint1 = 255;
+  live_speed->overall_speed = speed >> 1;
+  live_speed->speed1 = speed >> 1;
+}
+
+void close_partial_dir_edge_from(baldr::GraphReader& reader,
+                                 baldr::TrafficTile& tile,
+                                 uint32_t index,
+                                 double percent_along,
+                                 baldr::TrafficSpeed* current,
+                                 const std::string& edge_name,
+                                 const std::string& end_node,
+                                 const gurka::map& map) {
+  baldr::GraphId tile_id(tile.header->tile_id);
+  auto edge = std::get<0>(gurka::findEdge(reader, map.nodes, edge_name, end_node));
+  if (edge.Tile_Base() == tile_id && edge.id() == index) {
+    uint8_t subsegment = static_cast<uint8_t>(255 * percent_along);
+    SetLiveSpeedFrom(current, 0, subsegment);
+  }
+}
+
+void close_partial_dir_edge_upto(baldr::GraphReader& reader,
```

Comment:
Just an FYI:

 There are some utility functions already for customizing live traffic like `test::customize_live_traffic_data(map.config, cb_setter_max);` as used in https://github.com/valhalla/valhalla/blob/master/test/gurka/test_traffic.cc#L56

No

Reply:
Ohh I see that you are in fact using that below.

Reply:
yup :slightly_smiling_face: 

I added these functions here since I wanted partial closures. If this is reused elsewhere in the future, we could move it to the gurka module itself.